cachito/workers/pkg_managers/yarn.py

```diff
@@ -90,10 +117,23 @@ def _get_deps(yarn_lock, file_deps_allowlist):
     """
     deps = []
     nexus_replacements = {}
+    non_dev_deps = set()
+
+    for dep_type in ["dependencies", "peerDependencies", "optionalDependencies"]:
+        if dep_type not in package_json:
+            continue
+        for name, version in package_json[dep_type].items():
+            dep = f"{name}@{version}"
+            non_dev_deps = non_dev_deps.union(_find_reachable_deps(yarn_lock, dep))
```

Comment:
Why union and not just `.update()` ?  Why do you need create new instance every time and reallocate memory every time?

Reply:
Old habits! Yes, `update()` should be more efficient.